src/JsonApi/JsonSchema/SchemaFactory.php

```diff
@@ -244,8 +267,17 @@ private function getRelationship(string $resourceClass, string $property, ?array
                 continue;
             }
             $isRelationship = true;
+            $resourceMetadata = $this->resourceMetadataCollectionFactory->create($className);
+            $hasOperations ??= !$resourceMetadata->getOperation() instanceof NotExposed;
```

Comment:
I see, not a huge fan of using the `NotExposed` class, especially that the class is final, isn't there something else we could use like `output: false`? I'd like to avoid adding an interface if possible.

Reply:
`$operation->canRead()` does the trick! :+1: